graphql_api/types/commit/commit.py

```diff
@@ -190,12 +191,29 @@ def resolve_path_contents(commit: Commit, info, path: str = None, filters=None):
                 f"missing coverage for report with components: {component_filter}"
             )
 
-        commit_report = components.component_filtered_report(
-            commit_report, filtered_components
-        )
+        for component in filtered_components:
+            component_paths.extend(component.paths)
+
+        if flags_filter:
```

Comment:
what if there is no flags_filter but there are flags embedded in components? maybe you can do something like:

```
components_flags = []
 for component in filtered_components:
      components_flags.extend(
          component.get_matching_flags(commit_report.flags.keys())
      )
if components_flags:
  flags_filter = list(set(flags_filter) & set(component_flags))
```

just to make sure we're always handling the components flags

Reply:
Discussed this, we will be going with intersection query.

Reply:
The code i suggested doesn't disapprove the intersection query, we still need to take out the components_flags from the flags_filter condition block, this is only calling the components flags if filter_flags exist and we would want to avoid this